assets/src/edit-story/components/panels/document/status/status.js

```diff
@@ -128,31 +145,40 @@ function StatusPanel() {
   return (
     <SimplePanel
       name="status"
-      title={__('Status & Visibility', 'web-stories')}
+      title={__('Status and visibility', 'web-stories')}
       collapsedByDefault={false}
     >
       <>
+        <Row>
+          <HelperText>
+            {__('Set the current status of your story to', 'web-stories')}
+          </HelperText>
+        </Row>
         <Row>
           <RadioGroup
+            groupLabel="Visibility"
+            name="radio-group-visibility"
             options={visibilityOptions}
             onChange={handleChangeVisibility}
             value={getStatusValue(status)}
           />
         </Row>
         {passwordProtected === status && (
-          <>
-            <Row>
-              <BoxedTextInput
-                label={__('Password', 'web-stories')}
-                value={password}
-                onChange={handleChangePassword}
-                placeholder={__('Enter a password', 'web-stories')}
-              />
-            </Row>
-            <HelperText isWarning={password && password.length > 20}>
-              {__('Must not exceed 20 characters', 'web-stories')}
-            </HelperText>
-          </>
+          <InputRow>
+            <Input
+              aria-label={__('Password', 'web-stories')}
+              value={password}
+              onBlur={handleUpdatePassword}
+              onChange={handleChangePassword}
+              placeholder={__('Enter a password', 'web-stories')}
+              hasError={passwordError}
+              hint={
+                passwordError
+                  ? __('Must not exceed 20 characters', 'web-stories')
```

Comment:
⚠️  Warning: pet peeve incoming ⚠️ 

When we do enable this, I'd get rid of this restriction. It's a really stupid restriction and infuriates people more than it helps anyone. Except if someone can explain why shorter passwords are more secure?

Of course unless it's a WP restriction, in which case I just hate that restriction rather than this one 🤷

Reply:
I think we should just remove it now 🤷. If it ends up being needed we can add it later

Reply:
The only constraint in WP is the database field (255 chars). So yeah let's remove this limit here.